src/ioctls/vm.rs

```diff
@@ -610,6 +610,62 @@ impl VmFd {
         Ok(())
     }
 
+    /// Enable the specified capability as per the `KVM_ENABLE_CAP` ioctl.
+    ///
+    /// See the documentation for `KVM_ENABLE_CAP`.
+    ///
+    /// Returns an io::Error when the capability could not be enabled.
+    ///
+    /// # Arguments
+    ///
+    /// * kvm_enable_cap - KVM capability structure. For details check the `kvm_enable_cap`
+    ///                    structure in the
+    ///                    [KVM API doc](https://www.kernel.org/doc/Documentation/virtual/kvm/api.txt).
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # extern crate kvm_ioctls;
+    /// extern crate kvm_bindings;
+    ///
+    /// # use kvm_ioctls::{Cap, Kvm, VmFd};
+    /// use kvm_bindings::{kvm_enable_cap, KVM_CAP_SPLIT_IRQCHIP};
+    ///
+    /// let kvm = Kvm::new().unwrap();
+    /// let vm = kvm.create_vm().unwrap();
+    /// let mut cap: kvm_enable_cap = Default::default();
+    /// // This example cannot enable an arm/aarch64 capability since there
+    /// // is no capability available for these architectures.
+    /// if cfg!(target_arch = "x86") || cfg!(target_arch = "x86_64") {
+    ///     cap.cap = KVM_CAP_SPLIT_IRQCHIP;
+    ///     // As per the KVM documentation, KVM_CAP_SPLIT_IRQCHIP only emulates
+    ///     // the local APIC in kernel, expecting that a userspace IOAPIC will
+    ///     // be implemented by the VMM.
+    ///     // Along with this capability, the user needs to specify the number
+    ///     // of pins reserved for the userspace IOAPIC. This number needs to be
+    ///     // provided through the first argument of the capability structure, as
+    ///     // specified in KVM documentation:
+    ///     //     args[0] - number of routes reserved for userspace IOAPICs
+    ///     //
+    ///     // Because an IOAPIC supports 24 pins, that's the reason why this test
+    ///     // picked this number as reference.
+    ///     cap.args[0] = 24;
+    ///     vm.enable_cap(&cap).unwrap();
+    /// }
+    ///
+    /// ```
+    ///
+    pub fn enable_cap(&self, cap: &kvm_enable_cap) -> Result<()> {
```

Comment:
What architectures is this available on? The [documentation](https://www.kernel.org/doc/Documentation/virtual/kvm/api.txt) only mentions `mips, ppc, s390`, I found patches for x86, but what about aarch64?

Reply:
Oh my bad! Talking with @andreeaflorescu I realized this snippet of the doc is not part of 4.14 or 4.20, it's only been introduced later in 5.x.
So I suggest that for now I can remove the arm/aarch64 target.

Reply:
Sounds good to me.

Reply:
Ok I've updated the PR!